providers/fcm/src/lib/fcm.provider.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IPushOptions,
+  IPushProvider,
+} from '@novu/stateless';
+import { initializeApp, cert } from 'firebase-admin/app';
+import { getMessaging, Messaging } from 'firebase-admin/messaging';
+import crypto from 'crypto';
+
+export class FcmPushProvider implements IPushProvider {
+  id = 'fcm';
+  channelType = ChannelTypeEnum.PUSH as ChannelTypeEnum.PUSH;
+
+  private messaging: Messaging;
+  constructor(
+    private config: {
+      secretKey: string;
+    }
+  ) {
+    this.config = config;
+    const firebase = initializeApp(
+      {
+        credential: cert(JSON.parse(this.config.secretKey)),
```

Comment:
Here we are using the entirecredentials object, does the intializeApp a;also supports passing individual fields such as: secreyKey, publicKey and etc...? Just as it exists in the credentials JSON?

Reply:
It would be possible, but it would require the user to manually copy and paste the values from the JSON by hand.

Reply:
I think that for now let's go with separate fields by hand, and we can later add an autofiller based on a pasted JSON.